src/compiler/diagnosticMessages.json

```
@@ -5975,6 +5975,14 @@
         "category": "Error",
         "code": 7057
     },
+    "This syntax is reserved in files with this extension. Use an `as` expression instead.": {

```

This is slightly better, but only worthwhile if it's pretty easy:

```suggestion
    "This syntax is reserved in files with the extension {0}. Use an `as` expression instead.": {
```

(and for the other message too)